src/models/userModel.js

```diff
@@ -0,0 +1,44 @@
+import mongoose, { Schema } from "mongoose";
+
+const userSchema = new Schema({
+    name: {
+        type: String,
+        required: true
+    },
+    email: {
+        type: String,
+        required: true,
+        unique: true
+    }
+});
+
+const User = mongoose.model('user', userSchema);
+
+const userExists = async (email, session) => {
+    const user = await User.findOne({
+        email: email
+    }, [session]);
+
+    if(user !== null) {
+        return true;
+    }
+
+    return false;
```

Comment:
isso pode ser resumido em: 

```suggestion
    return !!user;
```

- https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Operators/Logical_NOT#duplo_not_!!